concourse/pipelines/guest-package-build.yaml

```diff
@@ -1596,6 +1596,489 @@ jobs:
+    task: failure
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "promote-windows-auto-updater-stable"
+      result_state: "failure"
+      start_timestamp: ((.:start-timestamp-ms))
+
+- name: upload-windows-powershell-staging
+  plan:
+  - in_parallel:
+      steps:
+      - get: compute-image-windows-tag
+        passed: [build-compute-image-windows]
+      - get: guest-test-infra
+      - get: compute-image-windows
+  - task: generate-timestamp
+    file: guest-test-infra/concourse/tasks/generate-timestamp.yaml
+  - load_var: start-timestamp-ms
+    file: timestamp/timestamp-ms
+  - load_var: package-version
+    file: compute-image-windows-tag/version
+  - task: generate-build-id
+    file: guest-test-infra/concourse/tasks/generate-build-id.yaml
+    vars:
+      prefix: ""
+  - load_var: build-id
+    file: build-id-dir/build-id
+  - task: upload-windows-powershell-to-staging
+    file: guest-test-infra/concourse/tasks/gcloud-package-operation.yaml
+    vars:
+      package_path: compute-image-windows/google-compute-engine-powershell.((.:package-version)).0@1.goo
+      universe: cloud-yuck
+      repo: google-compute-engine-powershell
+    params:
+      TYPE: uploadToStaging
+  on_success:
+    task: success
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "upload-windows-powershell-staging"
+      result_state: "success"
+      start_timestamp: ((.:start-timestamp-ms))
+  on_failure:
+    task: failure
+    file: guest-test-infra/concourse/tasks/publish-job-result.yaml
+    vars:
+      pipeline: "guest-package-build"
+      job: "upload-windows-powershell-staging"
+      result_state: "failure"
+      start_timestamp: ((.:start-timestamp-ms))
+
+- name: promote-windows-powershell-stable
+  plan:
+  - get: compute-image-windows-tag
+    passed: [upload-windows-powershell-staging]
+  - get: compute-image-windows
+    params: {fetch_tags: true}
+  - get: guest-test-infra
+  - task: generate-timestamp
+    file: guest-test-infra/concourse/tasks/generate-timestamp.yaml
+  - load_var: start-timestamp-ms
+    file: timestamp/timestamp-ms
+  - task: get-last-stable-date
+    file: guest-test-infra/concourse/tasks/get-last-stable-tag.yaml
+    input_mapping:
+      repo: compute-image-windows
+  - load_var: last-stable-date
+    file: last-stable-tag/date
+  - in_parallel:
+      - task: promote-windows-powershell-to-stable
+        file: guest-test-infra/concourse/tasks/gcloud-promote-package.yaml
+        vars:
+          universe: cloud-yuck
+          repo: google-compute-engine-powershell
+          environment: stable
+  - task: publish-time-since-stable
+    file: guest-test-infra/concourse/tasks/publish-time-since.yaml
+    vars:
+      last_publish_date: ((.:last-stable-date))
+      package_name: "google-compute-engine-powershell"
+  - put: compute-image-windows-tag
```

Comment:
this model won't work; you have copied the steps from other package builds, but they promote all packages from a repo at once and you've pulled them out. i think you should merge them back together, but at any rate this tagging scheme won't work as-is. you'd be overwriting the 'stable' tag with each individual package from the repo, whereas the tag refers to the repo as a whole.

Reply:
I've removed these steps.